components/ProfileLessons.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+
+type Lesson = {
+  order: number
+  progress?: number
+}
+
+type LessonsProp = {
+  lessons: Lesson[]
+}
+
+type lesson = {
```

Comment:
Do you think we need another type here or do you think that we can just check it lesson.progress doesn’t exist that we render the opacity, etc accordingly?

Reply:
> Do you think we need another type here or do you think that we can just check it lesson.progress doesn’t exist that we render the opacity, etc accordingly?

It will always have lessons.progress, i added that condition before it renders.

Reply:
Spoke with Rahul. To be discussed later, but I’m not sure if we should make another type if only one prop is different and the absence of the prop can be handled by JS. Should we maybe just assume if the lesson.progress key does not exist that we should just render accordingly?